src/HttpCache/EventListener/AddTagsListener.php

```diff
@@ -66,6 +73,16 @@ public function onKernelResponse(FilterResponseEvent $event)
             return;
         }
 
-        $event->getResponse()->headers->set('Cache-Tags', implode(',', $resources));
+        if ($this->debug) {
```

Comment:
I'm not sure it is worth it. The Formatter can do it by itself by using `RequestStack` (and it's useless for the default purger).

Reply:
By the way, even in your purger, I suggest to emit non-encoded headers, and to do the encoding/decoding at the cache store level.

Reply:
It's just an easy debugging help without any negative side effects. 
It's the Symfony Cache that does not allow certain characters, that comment should probably go there :-) I still think it's fine here. Sometimes you don't have control over the cache internals and then you're lost without that. I will make both optional using setter injection though.

Reply:
Sorry to insist but please drop it, it brings nothing for the default setup (cache tags are already in clear text) and may cause maintenance issues (I don't like injecting the `debug` parameter in this listener). It can easily be added in an external class/bundle.